### PR TITLE
Feature/memprof under benchmark

### PR DESCRIPTION
**Pull request by FranFin**
This PR groups the new sections into with the old `benchmarking` section has been split (`concept` and `result`) and the new `memo_prof` page under the same tab `benchmarking`.

### CODE REVIEW

**Review by ralph-lange — DISMISSED** [review]

**Comment by walkiry** [comment]
@FranFin Do you think it would be worth to mention within the benchmarking result the profiling results?

**Review by walkiry — APPROVED** [review]

**Comment by FranFin** [comment]
@walkiry You mean just a sentence referring to the Memory Profiling page? That's fine with me!

File: _docs/concepts/benchmarking/benchmarking/index.md [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+ * Then the output file will be added in a folder corresponding to the platform
+   type and benchmarking type.
+
+Being generic is the key for this benchmarking tool. The plugins will in
+contrary bring the specific implementation needed to benchmark  a specific
+platform. Every plugin will provide information as  requested by the parser:
+
+ * Provide a list of supported platforms. 
+ * Provide a list of functions that are handled.
+ * Provide snippets codes that will be added for benchmarking.
```
Comment (jamoralp) [inline _docs/concepts/benchmarking/benchmarking/index.md:72]:
```suggestion
 * Provide snippet codes that will be added for benchmarking.
```

File: _docs/concepts/benchmarking/benchmarking/index.md [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+ * Provide a list of patches and/or patch code
+ * Optional provide an end script to run and execute the benchmarks
+
+
+## Shadow Builder
+
+This section will introduce some concept related to the shadow builder (SB).
```
Comment (jamoralp) [inline _docs/concepts/benchmarking/benchmarking/index.md:79]:
```suggestion
This section will introduce some concepts related to the shadow builder (SB).
```

File: _docs/concepts/benchmarking/benchmarking/index.md [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+The Shadow builder is a tool that will transparently instrument the code to
+benchmark. The tools will be able to output an “instrumented code” that will be
+later be compiled as a normal code. The following steps describe what the shadow
+builder process flow:
+
+ * Get configuration file from the user (Benchmarking Configuration File).
+ * Get appropriate sources.
+ * Execute Trace Framework Abstraction Configuration file.
+ * Parse the sources file needed Injecting code.
+ * Compile the targeted binary for different platform.
```
Comment (jamoralp) [inline _docs/concepts/benchmarking/benchmarking/index.md:90]:
```suggestion
 * Compile the targeted binary for different platforms.
```

File: _docs/concepts/benchmarking/benchmarking/index.md [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+ * If needed, depending what type benchmark is undertaken, compile another
+   target binary benchmarking.
+
+The SB (Shadow Builder) is meant to be as transparent as possible for the user.
+And if the benchmarking is not activated, it should be bypassed.  Get
+appropriate sources
```
Comment (jamoralp) [inline _docs/concepts/benchmarking/benchmarking/index.md:96]: That sentence is incomplete
```suggestion
And if the benchmarking is not activated, it should be bypassed.
```

File: _docs/concepts/benchmarking/benchmarking/index.md [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+
+The SB is in charge of getting the path/git repository to the source code that
+needs to be benchmarking. The benchmarking. The sources are specified by the
+user in the benchmarking configuration file.  Injecting code
```
Comment (jamoralp) [inline _docs/concepts/benchmarking/benchmarking/index.md:100]:
```suggestion
user in the benchmarking configuration file.
```

**Review by jamoralp — CHANGES_REQUESTED** [review]

**Review by pablogs9 — APPROVED** [review]

**Review by amx-piap — DISMISSED** [review]

**Review by jamoralp — APPROVED** [review]

**Review by ralph-lange — APPROVED** [review]

**Pull request merged**